### PR TITLE
Add cmake files for linux build

### PR DESCRIPTION
**Pull request by noobusdeer**
add cinder/filesystem include for linux build

### CODE REVIEW

**Comment by notlion** [comment]
Hey thanks so much for this PR. Not sure why I didn't see it earlier. Sorry for the delay.

Currently the shared library build is not working for me. The samples build successfully after removing `SHARED` from the `add_library` command. I'd actually prefer if NanoVG is built as a static library, anyway. It's probably the more common case to link it into your application rather than having a .so floating around.

Other than that, if you can you fix up the mixed spaces and tabs used in the cmake files and remove the Windows font changes from this PR i'd love to get it merged in!

File: src/ci_nanovg.cpp [hunk 1]
```diff
@@ -344,7 +344,15 @@ void Context::stroke() {
 // Text //
 
 int Context::createFont(const std::string &name, const fs::path &filepath) {
+#if defined( CINDER_UWP ) || ( defined( _MSC_VER ) && ( _MSC_VER >= 1900 ) )
+	std::wstring ws(filepath);
+  file = std::string(ws.begin(), ws.end());
```
Comment (notlion) [inline src/ci_nanovg.cpp]: Curious. Why is the roundtrip through std::wstring and then std::string needed on windows? If it is, this should be submitted as a separate PR.

File: proj/cmake/NanoVGConfig.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
+if( NOT TARGET NanoVG )
+	get_filename_component( NanoVG_PATH "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE )
+
+	list( APPEND NanoVG_SOURCES
+		${NanoVG_PATH}/src/ci_nanovg.cpp
+		${NanoVG_PATH}/src/ci_nanovg_gl.cpp
+		${NanoVG_PATH}/src/SvgRenderer.cpp
+	)
+	add_library( NanoVG SHARED ${NanoVG_SOURCES} )
```
Comment (notlion) [inline proj/cmake/NanoVGConfig.cmake]: I think a static library build would be the better default.

File: proj/cmake/NanoVGConfig.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	        "$ENV{CINDER_PATH}/${CINDER_LIB_DIRECTORY}" )
+	endif()
+
+	target_include_directories( NanoVG PUBLIC "${NanoVG_PATH}/include" )
+	
+	set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+	get_filename_component( NanoVG_Deps_PATH "${NanoVG_PATH}/deps/nanovg/src" ABSOLUTE )
+	add_library( NanoVGDeps ${NanoVG_Deps_PATH}/nanovg.c )
+	target_include_directories( NanoVGDeps PUBLIC "${NanoVG_Deps_PATH}" )
+	add_dependencies( NanoVG NanoVGDeps )
```
Comment (notlion) [inline proj/cmake/NanoVGConfig.cmake]: Deps probably don't need their own build target. If people want to link against NanoVG without this wrapper they should just use the original project.

File: include/ci_nanovg.hpp [hunk 4]
```diff
@@ -47,6 +48,7 @@ class Context {
 
   void path2dSegment(Path2d::SegmentType, const vec2 *p1, const vec2 *p2, const vec2 *p3);
 
+  std::string file;
```
Comment (notlion) [inline include/ci_nanovg.hpp]: Looks like this does not need to be member data. It should just be a local variable in `Context::createFont()`.

**Review by notlion — CHANGES_REQUESTED** [review]
A few more specific comments.. Thanks again!

File: src/ci_nanovg.cpp (hunk 1 above)
Comment (noobusdeer) [reply src/ci_nanovg.cpp]: I made it, becouse on windows for filesystem work, cinder use not boost library as on linux an mac but they own library.
so fs::path 

#if defined( CINDER_UWP ) || ( defined( _MSC_VER ) && ( _MSC_VER >= 1900 ) )
	namespace fs = std::tr2::sys;
} // namespace cinder
#else
	namespace fs = boost::filesystem;
} // namespace cinder
